.github/workflows/test.yml

```diff
@@ -53,28 +73,168 @@ jobs:
         with:
           dotnet-version: "8.0.x"
+      - name: Cache NuGet packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj') }}
+          restore-keys: ${{ runner.os }}-nuget-
       - name: Run tests
         run: ./.ci/test.sh
         env:
           DOTNET_CONSOLE_ANSI_COLOR: true
+          
+  container-test-builds:
+    name: Build ${{matrix.tag}} Container
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - dockerfile: ConsumerApi/Dockerfile
+            tag: consumer-api
+          - dockerfile: AdminApi/src/AdminApi/Dockerfile
+            tag: admin-ui
+          - dockerfile: Modules/Devices/src/Devices.AdminCli/Dockerfile
+            tag: seed-client
+          - dockerfile: DatabaseMigrator/Dockerfile
+            tag: database-migrator
+          - dockerfile: EventHandlerService/src/EventHandlerService/Dockerfile
+            tag: event-handler-service
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          check-latest: true
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Setup Docker
+        uses: docker/setup-buildx-action@v3
+
+      - name: Setup NuGet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Cache Image
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-${{ matrix.tag }}-cache
+          key: buildx-${{ matrix.tag }}-cache-${{ hashFiles(matrix.dockerfile) }}
+          restore-keys: |
+            buildx-${{ matrix.tag }}-cache-
+            
+      - name: Install script dependencies
+        run: npm install --prefix ./.ci
+        
+      - name: Build Container
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ${{ matrix.dockerfile }}
+          cache-from: type=local,src=/tmp/.buildx-${{ matrix.tag }}-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-${{ matrix.tag }}-cache-new
+          tags: ${{ matrix.tag }}
+          load: false
+          outputs: type=docker,dest=/tmp/${{ matrix.tag }}.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: docker-${{ matrix.tag }}
+          path: /tmp/${{ matrix.tag }}.tar
+          retention-days: 1
+          
+      - name: Save Docker Logs
+        if: failure()
+        run: docker compose -f ./.ci/docker-compose.test.yml -f ./.ci/docker-compose.test.postgres.yml logs > docker-log.txt
+
+      - name: Archive logs
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-test-postgres-docker-logs
+          path: docker-log.txt
+        env:
+          DOTNET_CONSOLE_ANSI_COLOR: true
```

Comment:
Why are you saving docker compose logs when you didn't even do anything with compose in this job?